### DISCUSSION TITLE
Query different data by users base on same table

### DISCUSSION THREAD

**Discussion by razerzhang** [Budibase/budibase#6708] [Help] [answered]
I use MySQL as datasource, I want to make different users base on same role acquire different data.
for example: 
there is  a role  `manager` , `user1`  and `user2` belong to `manager`. both  `user`  want to get `budget` data which belong to themself.
I want to known whether  can I write sql to query `budget` table according to user current logged in like below pic?
And how can I bind user who current logged in with `budget` data?
![image](https://user-images.githubusercontent.com/76929114/179050702-00f06288-38ee-4367-be74-49865cf6d4b1.png)

**Comment by R2bEEaton** [comment, marked as answer]
Hey there!

### With handlebars in a query

You absolutely can. Follow the steps below and that should hopefully get you all set up.

1. In the SQL Query, press the "Add Binding" button. Set a binding name (like "Email") and a default value. The default value can be used for testing or be left blank.
2. In the Fields section, set up your query as before, except using the binding we just created.
```
SELECT * FROM `user`
WHERE `name` = {{ Email }};
```
3. At this point, you can click the buttons below to run the query with your default value to see if everything works. Then save the query.
4. Go to the Design tab and select or create a screen you want the data to be displayed on.
5. Create a new Data Provider and select your query from the list of data sources. After you do this, a gear icon should show up to the right of the data source picker.
![image](https://user-images.githubusercontent.com/34921506/179218160-5538115c-4434-49d6-b5fd-e7d645183616.png)
6. Click on the gear and enter the `{{ Current User.email }}` handlebar in the value box, then save.

### With filtering

Or, alternatively, you could use a data source filter depending on what you want to do. This will be slower if your query returns many rows.

1. Set up your query with no WHERE statement or bindings.
```
SELECT * FROM `users`;
```
2. Add a new data provider on a screen and choose your query from the list.
3. Click the "Define filters" button in the data source properties, then add a filter expression.
4. Choose the column you want to filter by ("name", or "Comment" in my example), "Equals", "Binding", `{{ Current User.email }}`.
![image](https://user-images.githubusercontent.com/34921506/179218918-6ec56acd-8a24-46e9-9292-0ff9ae1daf26.png)

**Reply by razerzhang** [reply]
thank u very much about these detailed explanation, I had try both ways and get result

**Reply by R2bEEaton** [reply]
You're very welcome! If this worked for you, please mark your question as answered.
